File: cs50 2019/pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    // cheking the input
    if (argc != 2) {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    for (int i = 0; i<strlen(argv[1]); i++) {
        if (!isdigit(argv[1][i])) {
            printf("Usage: ./caesar key\n");
            return 1;
        }
    }

    int key = atoi(argv[1]);
    string plaintext = get_string("plaintext: ");
    for (int i = 0; i<strlen(plaintext); i++) {
        if (isupper(plaintext[i])) {
            printf("%c", ((plaintext[i]-65 + key) % 26) + 65);
        } else if (islower(plaintext[i])) {
            printf("%c", ((plaintext[i]-97 + key) % 26) + 97);
        } else {
            printf("%c", plaintext[i]);
        }
    }
    printf("\n");
}

File: cs50 2019/pset6/caesar.py
from sys import argv, exit
from cs50 import get_string

k = int(argv[1])
if k < 0 or len(argv) != 2:
    exit(1)

plaintext = get_string("plaintext: ")
print("ciphertext: ", end='')

for i in plaintext:
    if i.isupper():
        print(chr(((ord(i) - 65 + k) % 26) + 65), end='')
    elif i.islower():
        print(chr(((ord(i) - 97 + k) % 26) + 97), end='')
    else:
        print(i, end='')
print()
